xskillscore/core/np_deterministic.py

```diff
@@ -127,15 +150,18 @@ def _pearson_r_p_value(a, b, weights, axis, skipna):
     scipy.stats.pearsonr
 
     """
+    if skipna:
+        a, b, weights = _match_nans(a, b, weights)
     r = _pearson_r(a, b, weights, axis, skipna)
     if np.isnan(r).all():
         return r
     else:
         # no nans or some nans
         a = np.rollaxis(a, axis)
         b = np.rollaxis(b, axis)
-        dof = np.apply_over_axes(np.sum, np.isnan(a * b), 0).squeeze() - 2
-        dof = np.where(dof > 1.0, dof, a.shape[0] - 2)
```

Comment:
These two lines were confusing to me and were breaking cases with nans. The below code `np.count_nonzero` is a straight forward way to get at the degrees of freedom.

I imagine line 138 was to deal with cases with 0 dof? But it replaced 0 or less dof with the max dof in this case? I don't fully understand, but it seemed to work anyways. So with the method below we don't get `dof < 1` cases to return `1.0` for the p-value like in `scipy.stats.pearsonr`, but instead it just returns a nan.